### PR TITLE
feat(format): support mapped VALUE suggester display text

### LINKED ISSUE #588: [FEATURE REQUEST] Support "text item" and "item" in suggester
**Is your feature request related to a problem? Please describe.**
Maybe this exists and I just can't find it. I have a "Capture".  I want the suggestions displayed to be different from the actual values used.

I have a capture format which contains "{{VALUE: ,🔼,⏫,🔽,⏬}}".  What I'd like is something like "{{VALUE:"3-Normal", "2-High", "1-Urgent", "4-Low", "5-Why Bother"; ,🔼,⏫,🔽,⏬}}.  I want the numbered text items to be displayed, but the value inserted in the capture to be the corresponding items in the second list.  I used a semicolon to separate the two lists, but that doesn't have to be the syntax.  I just want two lists; one for the display and one to be inserted in the capture

**Describe the solution you'd like**

**Additional context**

### PR DESCRIPTION
**Pull request by chhoumann**
Support separate suggester display text and inserted values for VALUE option lists.

This adds `|text:` support to `{{VALUE:...}}` so users can show friendly labels in the suggester while storing/inserting a different mapped value.

What changed:
- Extend VALUE parsing with `|text:` and strict validation:
  - only valid for option-list VALUE tokens
  - display and item counts must match
  - display entries must be unique
- Runtime formatting now renders mapped display values while inserting mapped item values.
- Preflight One Page Input flow now preserves the same mapping behavior:
  - dropdown shows display labels and stores mapped item values
  - suggester maps only completion selections; typed custom input remains typed
- Add focused tests for parser/runtime/preflight mapping paths.
- Document new syntax and current limitations in Format syntax docs.

This addresses the feature request to decouple display from inserted values in VALUE suggesters.

Fixes #588

Verification:
- `bun run test`
- `bun run lint`
- `bun run build`
- `obsidian vault=dev plugin:reload id=quickadd`
- `obsidian vault=dev eval ...` edge-case pack (7/7 pass)

---

<a href="https://app.devin.ai/review/chhoumann/quickadd/pull/1127" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open with Devin
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * VALUE text mapping: option lists can show custom labels while inserting different underlying values; dropdowns and autocomplete now display those labels and map selections back to stored values.

* **Bug Fixes / Validation**
  * Added validation and errors for mismatched counts and duplicate display labels in VALUE mappings.

* **Documentation**
  * Format syntax docs updated with the new VALUE text mapping variant and usage constraints.

### CODE REVIEW

File: src/utils/valueSyntax.ts [hunk 1]
```diff
@@ -160,6 +170,9 @@ function parseOptions(optionParts: string[], hasOptions: boolean): ParsedOptions
 			case "type":
 				if (value) inputTypeOverride = value;
 				break;
+			case "text":
+				displayValuesRaw = value;
+				break;
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/utils/valueSyntax.ts:175]: Implemented in df0cb69. I validated this path and added explicit enforcement in `parseAnonymousValueOptions` so `{{VALUE|text:...}}` now throws the same option-list-only error as other single-value forms. Added regression coverage in `src/utils/valueSyntax.test.ts` (`throws when text mappings are used on unnamed VALUE tokens`).

File: src/utils/valueSyntax.ts [hunk 2]
```diff
@@ -5,7 +5,14 @@ export const VALUE_LABEL_KEY_DELIMITER = "\u001F";
...
 
-const VALUE_OPTION_KEYS = new Set(["label", "default", "custom", "type", "case"]);
+const VALUE_OPTION_KEYS = new Set([
+	"label",
+	"default",
+	"custom",
+	"type",
+	"case",
+	"text",
+]);
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/utils/valueSyntax.ts:15]: Validated this behavior against current design and docs. We are intentionally reserving `|text:` for option-list VALUE tokens only and failing fast on single-value/anonymous forms (including `{{VALUE:name|text:foo}}` and `{{VALUE|text:foo}}`) rather than treating `text:...` as shorthand default text. This aligns runtime/parser behavior and the new FormatSyntax docs.
